src/datasets/load.py

```diff
@@ -408,11 +483,31 @@ def infer_module_for_data_files_in_archives(
     return None, {}
 
 
+def update_hash_with_config_parameters(hash: str, config_parameters: dict) -> str:
```

Comment:
`DatasetBuilder` also hashes the config params. This cannot be avoided?

Reply:
I think it only hashes the config params that the user pass to `load_dataset` - not the ones defined in a BuilderConfig instance in BUILDER_CONFIGS